core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -2725,6 +2696,22 @@ trueLiteral, literal(1),
     checkSimplifyUnchanged(divideNode4);
   }
 
+  @Test void testSimplifyIsNullDivideByZero() {
+    simplify = simplify.withParanoid(false);
+
+    RelDataType intType =
+        typeFactory.createTypeWithNullability(
+            typeFactory.createSqlType(SqlTypeName.INTEGER), false);
+
+    checkSimplifyUnchanged(isNull(div(vIntNotNull(), literal(0))));
+    checkSimplifyUnchanged(isNull(div(vIntNotNull(), cast(literal(0), intType))));
+    checkSimplify(isNull(div(vIntNotNull(), cast(literal(2), intType))), "false");
```

Comment:
How about adding also the following: `isNull(div(cast(literal(2), intType), vIntNotNull()))` to protect against possible regressions in the future.